.github/workflows/main.yml

```diff
@@ -22,6 +22,8 @@ jobs:
     runs-on: ${{ matrix.platform }}
     env:
       SETUPTOOLS_USE_DISTUTILS: ${{ matrix.distutils }}
+      PY_IGNORE_IMPORTMISMATCH: ${{ contains(matrix.python, 'pypy') && 1 }}
```

Comment:
Hi @kloczek, there is a problem with pytest that might be causing the problems you are seeing:
pytest-dev/pytest#2042

This seems to be related to the contents of your log.
I think you can try PY_IGNORE_IMPORTMISMATCH=1 as an env var to see how it goes.

Reply:
> I think you can try PY_IGNORE_IMPORTMISMATCH=1 as an env var to see how it goes.

Tested. Result with `pytest` 6.2.5 is exactly the same.
```console
+ cd setuptools-60.5.4
+ export PY_IGNORE_IMPORTMISMATCH=1
+ PY_IGNORE_IMPORTMISMATCH=1
+ PYTHONDONTWRITEBYTECODE=1
+ PYTHONPATH=/home/tkloczko/rpmbuild/BUILDROOT/python-setuptools-60.5.4-2.fc35.x86_64/usr/lib/python3.8/site-packages
+ pytest -ra -p no:randomly setuptools -W ignore::DeprecationWarning
=========================================================================== test session starts ============================================================================
platform linux -- Python 3.8.12, pytest-6.2.5, py-1.11.0, pluggy-1.0.0
rootdir: /home/tkloczko/rpmbuild/BUILD/setuptools-60.5.4, configfile: pytest.ini
plugins: shutil-1.7.0, forked-1.4.0, cov-3.0.0, virtualenv-1.7.0, checkdocs-2.7.1, flake8-1.0.7, xdist-2.5.0
collected 155 items / 20 errors / 1 skipped / 134 selected

================================================================================== ERRORS ==================================================================================
___________________________________________________________ ERROR collecting setuptools/tests/test_bdist_egg.py ____________________________________________________________
ImportError while importing test module '/home/tkloczko/rpmbuild/BUILD/setuptools-60.5.4/setuptools/tests/test_bdist_egg.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
setuptools/tests/test_bdist_egg.py:11: in <module>
    from . import contexts
E   ImportError: attempted relative import with no known parent package
___________________________________________________________ ERROR collecting setuptools/tests/test_build_ext.py ____________________________________________________________

[..]

========================================================================= short test summary info ==========================================================================
SKIPPED [1] setuptools/tests/test_msvc.py:17: could not import 'distutils.msvc9compiler': No module named 'winreg'
ERROR setuptools/tests/test_bdist_egg.py
ERROR setuptools/tests/test_build_ext.py
ERROR setuptools/tests/test_build_meta.py
ERROR setuptools/tests/test_config.py
ERROR setuptools/tests/test_depends.py
ERROR setuptools/tests/test_develop.py
ERROR setuptools/tests/test_dist.py
ERROR setuptools/tests/test_dist_info.py
ERROR setuptools/tests/test_easy_install.py
ERROR setuptools/tests/test_egg_info.py
ERROR setuptools/tests/test_install_scripts.py
ERROR setuptools/tests/test_msvc.py
ERROR setuptools/tests/test_namespaces.py
ERROR setuptools/tests/test_packageindex.py
ERROR setuptools/tests/test_sdist.py
ERROR setuptools/tests/test_test.py
ERROR setuptools/tests/test_upload_docs.py
ERROR setuptools/tests/test_virtualenv.py
ERROR setuptools/tests/test_wheel.py
ERROR setuptools/tests/integration/test_pip_install_sdist.py
!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!! Interrupted: 20 errors during collection !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
====================================================================== 1 skipped, 20 errors in 2.82s =======================================================================

Reply:
The [logs](https://github.com/pypa/setuptools/issues/2318#issuecomment-1016606068) seem very different for me 😄, looks like you are making progress!

Can you confirm that you are using the exact same flags used in setuptools own `tox.ini` file and /or `.github/workflows/main.yml`?

Can you also confirm that you are running pytest from a directory with all the configuration files present in the repository (`pyproject.toml`, `pytest.ini`, `setup.cfg`, `conftest.py`)?